File: ex11.py
class Solution:
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """

        max_area = 0       
        head = 0
        rear = len(height) - 1


        while rear - head >= 1:
            if (rear - head) * min(height[head], height[rear]) > max_area:
                max_area = (rear - head) * min(height[head], height[rear])


            if height[head] > height[rear]:
                rear -= 1
            else:
                head += 1
        
            
        return max_area


sol = Solution()
print(sol.maxArea([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]))

File: code/ex11.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int n = height.size();
        int MAX = 0;
        
        int lo = 0, hi = n - 1;
        while (lo < hi) {
            MAX = max(MAX, (hi - lo) * min(height[lo], height[hi]));
            if (height[lo] < height[hi])    lo++;
            else    hi--;
        }
        
        return MAX;
    }
};